src/PAModel/CanvasDocument.cs

```diff
@@ -180,6 +180,10 @@ private static CanvasDocument Wrapper(Func<CanvasDocument> worker, ErrorContaine
             }
             catch (Exception e)
             {
+                if (throwOnException && !(e is DocumentException))
+                {
+                    throw;
+                }
```

Comment:
this could just catch DocumentExceptions and let the others pass by, if we aren't allowing the caller to control the behavior. 
```
try { ...
}
catch (DocumentException e)
{
// swallow, use existing behavior
}
```